app/models/user.rb

```diff
@@ -20,45 +18,17 @@ class User < ApplicationRecord
       attr_accessor :_skip_creating_profile
 
       # All new users should automatically have a profile
-      after_create_commit :create_profile, unless: :_skip_creating_profile
+      after_create_commit -> { Profile.create(user: self, data: Profiles::ExtractData.call(self)) },
```

Comment:
Inlining this for now to keep this inline concern short. If we still need this in a bit we can move it into the class proper and give it a dedicated method again if we feel like it.